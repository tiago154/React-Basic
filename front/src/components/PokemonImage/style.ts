import styled from 'styled-components';

interface IPropsStyled {
    width: number
}

const ImagePokemon = styled.img`
    ${({ width }: IPropsStyled) => width && `max-width: ${width}px`}
    height: auto;
`;

export default ImagePokemon