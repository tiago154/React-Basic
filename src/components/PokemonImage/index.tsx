import React from 'react'
import ImagePokemon from './style'

interface IProps {
  url: string,
  width: number
}

const PokemonImage = ({ url, width }: IProps) => (
  <ImagePokemon src={url} width={width} />
)

export default PokemonImage
