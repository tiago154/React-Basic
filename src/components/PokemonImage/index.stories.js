import React from 'react'
import PokemonImage from '.'
import { withKnobs, select, number } from '@storybook/addon-knobs'

export default {
  title: 'PokemonImage',
  decorators: [withKnobs]
}

const options = {
  charmander: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/004.png',
  bulbasaur: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png',
  squirtle: 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/007.png'
}

export const Imagens = () => <PokemonImage
  url={select('url', options, 'https://assets.pokemon.com/assets/cms2/img/pokedex/full/007.png')}
  width={number('width', 150)}
/>
