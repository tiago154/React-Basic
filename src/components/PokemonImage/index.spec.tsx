import React from 'react'
import PokemonImage from '.'
import renderer from 'react-test-renderer'

describe('PokemonImage', () => {
  it('Renders correctly', () => {
    const component = <PokemonImage
      url={'https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png'}
      width={120}
    />
    const tree = renderer
      .create(component)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
