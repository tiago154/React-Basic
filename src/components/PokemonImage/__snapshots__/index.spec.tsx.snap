// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonImage Renders correctly 1`] = `
<img
  className="sc-bdVaJa bPNabn"
  src="https://assets.pokemon.com/assets/cms2/img/pokedex/full/001.png"
  width={120}
/>
`;
